src/optimizer/remove_unused_columns.cpp

```diff
@@ -461,14 +693,76 @@ void BaseColumnPruner::AddBinding(BoundColumnRefExpression &col, ColumnIndex chi
 		ReferencedColumn column;
 		column.bindings.push_back(col);
 		column.child_columns.push_back(std::move(child_column));
-		column_references.insert(make_pair(col.binding, std::move(column)));
+		entry = column_references.emplace(make_pair(col.binding, std::move(column))).first;
 	} else {
 		// column reference already exists - check add the binding
 		auto &column = entry->second;
 		column.bindings.push_back(col);
 
 		MergeChildColumns(column.child_columns, child_column);
 	}
+	if (mode == BaseColumnPrunerMode::DISABLE_PUSHDOWN_EXTRACT) {
+		//! Any child referenced after this mode is set disables PUSHDOWN_EXTRACT
+		entry->second.supports_pushdown_extract = PushdownExtractSupport::DISABLED;
+	}
+}
+
+void ReferencedColumn::AddPath(const ColumnIndex &path) {
+	if (child_columns.empty()) {
+		//! Full field already referenced, won't use struct field projection pushdown at all
+		return;
+	}
+	path.VerifySinglePath();
```

Comment:
Maybe this could be extracted from the columns generated with `MergeChildColumns` instead of doing this again?

Reply:
I thought this would be a good optimization
but then I started working on pushing down the cast as well and this falls apart

If we have two identical paths but with a different cast on top, those need to be unique columns